src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -313,6 +313,43 @@ class C
                 Diagnostic(ErrorCode.ERR_AssgReadonly, "_ro").WithLocation(15, 47));
         }
 
+        [Fact, WorkItem(42259, "https://github.com/dotnet/roslyn/issues/42259")]
+        public void RefReturnLocalFunction()
+        {
+            var source = @"
+#pragma warning disable CS8321
+class C {
+    static void M(){
+        ref int M1(in int i) => ref i;
+        ref int M2(in int i) { return ref i; }
+        ref readonly int M3(in int i) => ref i;
+        ref readonly int M4(in int i) { return ref i; }
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                    // (5,37): error CS8333: Cannot return variable 'in int' by writable reference because it is a readonly variable
+                    //         ref int M1(in int i) => ref i;
+                    Diagnostic(ErrorCode.ERR_RefReturnReadonlyNotField, "i").WithArguments("variable", "in int").WithLocation(5, 37),
+                    // (6,43): error CS8333: Cannot return variable 'in int' by writable reference because it is a readonly variable
+                    //         ref int M2(in int i) { return ref i; }
+                    Diagnostic(ErrorCode.ERR_RefReturnReadonlyNotField, "i").WithArguments("variable", "in int").WithLocation(6, 43));
```

Comment:
>); [](start = 132, length = 2)

Consider moving ```);``` on a new line. Makes it easier to add/remove expected diagnostics in the future. Also, it looks like the lines we shifted to the right on an extra tab in the 3rd commit. We usually use single tab (4 spaces) offset.